docs/book/developer-guide/steps-pipelines/post-execution-workflow.md

```diff
@@ -0,0 +1,151 @@
+---
+description: How to inspect a finished pipeline run.
+---
+
+# Inspecting Pipeline Runs
```

Comment:
One thing missing here is the infrastructure/stack level of this stuff. We need to make clear that the metadata and artifact store is the critical part in post-execution. If you switch between stacks, you wont have access to the pipelines. I think this needs a small diagram here where we should that the repository object queries the metadata store (and the zenstores?) and creates these views. Then we .read() is called the artifact store is called directly. We also need to tell them that on the .read() the materializer is basically used again, so we can use any artifact store and fetch it locally.


Users need to see these to see that ZenML takes care of this for them but they need to know this. Its important that clients know also the permissions they need to make the function calls. E.g. Some users may have access to the metadata store but no the artifact store so they need to know how the calls work.

It is important also that the metadata store is exposed locally in the right ports to make this work! They should know that is what `zenml stack up` does basically.

Finally, they should know that this is different to the flow they might see when they try to fetch artifacts within a step :-)